src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/SearchParameterUtilities.cs

```diff
@@ -0,0 +1,40 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading;
+using EnsureThat;
+using Hl7.Fhir.ElementModel;
+using Microsoft.Health.Fhir.Core.Features.Definition;
+using Microsoft.Health.Fhir.Core.Features.Definition.BundleWrappers;
+using Microsoft.Health.Fhir.Core.Features.Search.Registry;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search.Parameters
+{
+    public class SearchParameterUtilities : ISearchParameterUtilities
+    {
+        private readonly ISearchParameterDefinitionManager _searchParameterDefinitionManager;
+        private readonly SearchParameterStatusManager _searchParameterStatusManager;
+
+        public SearchParameterUtilities(
+            SearchParameterStatusManager searchParameterStatusManager,
+            ISearchParameterDefinitionManager searchParameterDefinitionManager)
+        {
+            EnsureArg.IsNotNull(searchParameterStatusManager, nameof(searchParameterStatusManager));
+            EnsureArg.IsNotNull(searchParameterDefinitionManager, nameof(searchParameterDefinitionManager));
+
+            _searchParameterStatusManager = searchParameterStatusManager;
+            _searchParameterDefinitionManager = searchParameterDefinitionManager;
+        }
+
+        public async System.Threading.Tasks.Task AddSearchParameterAsync(ITypedElement searchParam, CancellationToken cancellationToken)
+        {
+            _searchParameterDefinitionManager.AddNewSearchParameters(new List<ITypedElement>() { searchParam });
+
+            var searchParameterWrapper = new SearchParameterWrapper(searchParam);
+            await _searchParameterStatusManager.AddSearchParameterStatusAsync(new List<string>() { searchParameterWrapper.Url });
```

Comment:
>AddSearchParameterStatusAsync [](start = 48, length = 29)

Just curious, do we plan in future to work with multiple parameters? Right now it feels like it's always one parameter only, and if that the case, we can simplified code if we move away from collection to single entity.

Reply:
We do plan to handle batches of SearchParameters soon.  Just starting with single add for the moment.